### PR TITLE
all: make optional fields omitempty

### PR DESCRIPTION
**Pull request by mhilton**
Also make time fields *time.Time.

### CODE REVIEW

**Review by fabricematrat — APPROVED** [review]

File: openapi/types.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ func TypeString(typ string, format string) string {
 		return "float64"
 	case "string":
 		if format == "date-time" {
-			return "time.Time"
+			return "*time.Time"
```
Comment (rogpeppe) [inline openapi/types.go:18]: I guess it would be nice not to use `*time.Time` when a field is required, but I'm guessing that would be too much hassle.

**Review by rogpeppe — APPROVED** [review]
LGTM thanks.

**Review by rogpeppe — APPROVED** [review]
LGTM thanks

**Comment by mhemmings** [comment]
@mhilton Thanks for this! LGTM.

(Note: this closes #18)

**Pull request merged**